controller/src/main/java/org/jboss/as/controller/ModelControllerImpl.java

```diff
@@ -449,6 +450,19 @@ boolean boot(final List<ModelNode> bootList, final OperationMessageHandler handl
             }
 
             resultAction = postExtContext.executeOperation();
+
+            if (!skipModelValidation && resultAction == OperationContext.ResultAction.KEEP && bootOperations.postExtensionOps != null) {
+                //Get the modified resources from the initial operations and add to the resources to be validated by the post operations
+                Set<PathAddress> validateAddresses = new HashSet<PathAddress>();
+                validateAddresses.addAll(context.getModifiedResourcesForModelValidation());
+                validateAddresses.addAll(postExtContext.getModifiedResourcesForModelValidation());
+                final AbstractOperationContext validateContext = new OperationContextImpl(operationID, POST_EXTENSION_BOOT_OPERATION,
+                        EMPTY_ADDRESS, this, processType, runningModeControl.getRunningMode(),
+                        contextFlags, handler, null, managementModel.get(), control, processState, auditLogger,
+                                bootingFlag.get(), hostServerGroupTracker, null, null, notificationSupport, false);
+                resultAction = validateContext.executeOperation();
+
+            }
```

Comment:
I don't get this part. It doesn't add any steps so I guess it's just supposed to validate. But the 'validateAddresses' set isn't used.